src/utils/recorder_events.jl

```diff
@@ -48,35 +48,59 @@ end
 
 struct InitialConditionUpdateEvent <: IS.AbstractRecorderEvent
     common::IS.RecorderEventCommon
-    simulation_time::Dates.DateTime
+    execution_timestamp::Dates.DateTime
     initial_condition_type::String
     device_type::String
     device_name::String
-    val::Float64
+    new_value::Float64
     previous_value::Float64
     problem_number::Int
 end
 
 function InitialConditionUpdateEvent(
     simulation_time,
-    key::ICKey,
     ic::InitialCondition,
-    val::Float64,
     previous_value::Float64,
     problem_number::Int,
 )
     return InitialConditionUpdateEvent(
         IS.RecorderEventCommon("InitialConditionUpdateEvent"),
         simulation_time,
-        string(get_entry_type(key)),
-        string(get_component_type(key)),
+        string(get_ic_type(ic)),
+        string(get_component_type(ic)),
         get_component_name(ic),
-        val,
+        get_condition(ic),
         previous_value,
         problem_number,
     )
 end
 
+struct ParameterUpdateEvent <: IS.AbstractRecorderEvent
+    common::IS.RecorderEventCommon
+    execution_timestamp::Dates.DateTime
+    parameter_type::String
+    device_type::String
+    name::String
```

Comment:
This field name may need to change. It is only valid if all ParameterAttributes subtypes will define `name`. We could call it something generic like `meta` or `tag`.

Reply:
`timeseries_name`?

Reply:
Note that I changed the field from `name` to `tag`. The base recorder event already uses `name` for the event type.

Using `time_series_name` implies that the event is only for TimeSeriesParameters, but that isn't right case, right? Won't we soon add events for other parameter types? I would be OK changing this event to be TimeSeriesParameterUpdateEvent. A consequence of doing that is that we would not be able to print all parameter update events in the same table.